packages/devextreme/js/__internal/ui/html_editor/ui/aiDialog.ts

```diff
@@ -1,33 +1,46 @@
 import '@js/ui/drop_down_button';
 
-import type { AIIntegration } from '@js/common/ai-integration';
+import type { AIIntegration, RequestCallbacks } from '@js/common/ai-integration';
 import localizationMessage from '@js/common/core/localization/message';
 import type { dxElementWrapper } from '@js/core/renderer';
 import $ from '@js/core/renderer';
 import { extend } from '@js/core/utils/extend';
 import type { ButtonClickEvent, ItemClickEvent } from '@js/ui/drop_down_button_types';
-import type { AICustomCommand } from '@js/ui/html_editor';
+import type { AICommandNameExtended, AICustomCommand } from '@js/ui/html_editor';
+import LoadIndicator from '@js/ui/load_indicator';
 import type { Properties as PopupProperties, ToolbarItem } from '@js/ui/popup';
 import type dxSelectBox from '@js/ui/select_box';
 import type { Properties as SelectBoxProperties } from '@js/ui/select_box';
 import SelectBox from '@js/ui/select_box';
 import TextArea from '@js/ui/text_area';
+import type {
+  AICommandExecutor,
+  AICommandParamsMap,
+  AICommandResultMap,
+  CommandDefinition,
+  CommandsMap,
+} from '@ts/ui/html_editor/utils/ai';
+import { buildAICommandParams, getAICommandName } from '@ts/ui/html_editor/utils/ai';
+import { TEXTEDITOR_INPUT_CONTAINER_CLASS } from '@ts/ui/text_box/m_text_editor.base';
 
-import type { CommandDefinition, CommandsMap } from '../utils/ai';
 import BaseDialog from './m_baseDialog';
 
-const AI_DIALOG_COMMANDS_WITH_OPTIONS = ['translate', 'changeStyle', 'changeTone', 'custom'];
-
 export const AI_DIALOG_CLASS = 'dx-aidialog';
 export const AI_DIALOG_CONTROLS_CLASS = 'dx-aidialog-controls';
 export const AI_DIALOG_CONTENT_CLASS = 'dx-aidialog-content';
+
+const AI_DIALOG_LOAD_INDICATOR_CLASS = 'dx-pending-indicator';
 const AI_DIALOG_TITLE_CLASS = 'dx-aidialog-title';
 const AI_DIALOG_TITLE_TEXT_CLASS = 'dx-aidialog-title-text';
 const ICON_CLASS = 'dx-icon';
 const ICON_SPARKLE_CLASS = 'dx-icon-sparkle';
 const COPY_BUTTON_ICON = 'copy';
 const TRY_AGAIN_BUTTON_ICON = 'restore';
 
+const AI_DIALOG_COMMANDS_WITH_OPTIONS = ['translate', 'changeStyle', 'changeTone'];
+const AI_DIALOG_ASKAI_COMMAND_NAME = 'askAI';
```

Comment:
Please fix width of "generate" button - not text is cut in Material theme
![image](https://github.com/user-attachments/assets/f45a8805-1b8b-4ab0-a8cb-2bc79d9d0e44)

Reply:
[06752a6977ad6f98e72d58b33163714847ce7454]

Reply:
It's sad to me that we are forced to use 109 in material specifically, cannot we just increate BUTTON_WIDTH?